### PR TITLE
[openblas] Support arm64-uwp builds

### LINKED ISSUE #32066: [openblas] Support arm64-uwp builds
### Is your feature request related to a problem? Please describe.

Support building OpenBLAS for ARM64 UWP. 

### Proposed solution

The OpenBLAS portfile.cmake can be updated to not explicitly set CMAKE_SYSTEM_PROCESSOR to AMD64.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by daniel-ju-msft**
Fixes #32066.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by Cheney-W — DISMISSED** [review]

**Comment by daniel-ju** [comment]
Can this be merged? Thanks!

File: ports/openblas/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "openblas",
   "version": "0.3.23",
+  "port-version": 1,
   "description": "OpenBLAS is an optimized BLAS library based on GotoBLAS2 1.13 BSD version.",
   "homepage": "https://github.com/xianyi/OpenBLAS",
   "license": "BSD-3-Clause",
-  "supports": "!(arm & uwp)",
+  "supports": "!((arm & !arm64) & uwp)",
```
Comment (BillyONeal) [inline ports/openblas/vcpkg.json:8]:
```suggestion
  "supports": "!arm | arm64 | !uwp",
```

Proof:

```
!((arm & !arm64) & uwp)    given
(!(arm & !arm64) | !uwp)   demorgan
((!arm | arm64) | !uwp)    demorgan
!arm | arm64 | !uwp        disjunctions are commutative and associative
```

File: ports/openblas/portfile.cmake [hunk 2]
```diff
@@ -42,9 +42,14 @@ endif()
 set(OPENBLAS_EXTRA_OPTIONS)
 # for UWP version, must build non uwp first for helper
 # binaries.
-if(VCPKG_TARGET_IS_UWP)    
-    list(APPEND OPENBLAS_EXTRA_OPTIONS -DCMAKE_SYSTEM_PROCESSOR=AMD64
-                "-DBLASHELPER_BINARY_DIR=${CURRENT_HOST_INSTALLED_DIR}/tools/${PORT}")
+if(VCPKG_TARGET_IS_UWP)
+    if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+        list(APPEND OPENBLAS_EXTRA_OPTIONS -DCMAKE_SYSTEM_PROCESSOR=ARM64
```
Comment (BillyONeal) [inline ports/openblas/portfile.cmake:47]: This seems like the default value of `CMAKE_SYSTEM_PROCESSOR`? Maybe the correct fix is just to stop trying to mess with that?

**Comment by BillyONeal** [comment]
Proposal: https://github.com/BillyONeal/vcpkg/commits/openblas-proposal

**Review by BillyONeal — APPROVED** [review]

**Comment by daniel-ju** [comment]
Thanks @BillyONeal for the review and improvements here!

**Pull request merged**